### PR TITLE
Add `dotnet format` to our CI

### PR DESCRIPTION
**Pull request by AnPucel**
## Description
Adding a linter to the CI. runs `dotnet format` and verifies nothing would be changed. 

## How has this been tested?
- Checked the CI

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+jobs:
+  lint_csharp:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup .NET Core
+        uses: actions/setup-dotnet@v2
+        with:
+          dotnet-version: "6.0.x"
```
Comment (dhuckins) [inline .github/workflows/lint.yml:18]: what version of dotnet should we actually use?
https://github.com/hashicorp/vault-client-dotnet/blob/main/.github/workflows/release.yml#L12

Comment (AnPucel) [reply .github/workflows/lint.yml:18]: the `dotnet format` tool was only available as of `5.0.0` and up.

wrt to releases and testing, we only use .Net for the build and test tooling so it doesn't really matter as long as it supports what we need. Likely we should change this all to latest `.Net 6`. 

When we build the library the version of .net is `.net standard`, which means that it's compatible with a good chunk of frameworks.

Comment (AnPucel) [reply .github/workflows/lint.yml:18]: I'll change them in release and testing to make sure they're more consistent.

**Comment by AnPucel** [comment]
Closing in favor of adding format to our makefile

**Pull request closed without merging**